From 58fca64a57013736a42a08db40fd02d4579a6d93 Mon Sep 17 00:00:00 2001
From: Pavan Kumar S <pavan.kumar.s@intel.com>
Date: Thu, 21 May 2015 08:20:35 +0530
Subject: [PATCH] [TYPE_C]: Adding cable name for DFP, UFP, DP Source and Sink
 in extcon

In Type-C, as we have different types of notification to handle based on the
functionality, added extcon cable names for UFP, DFP, DP Source and DP Sink.

Change-Id: Ie814d8f815c131d5a0eee746d53462fe3d891cda
Tracked-on: https://jira01.devtools.intel.com/browse/IMINAN-34498
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
Signed-off-by: Pavan Kumar S <pavan.kumar.s@intel.com>
---
 drivers/extcon/extcon-class.c | 5 ++++-
 include/linux/extcon.h        | 5 ++++-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/extcon/extcon-class.c b/drivers/extcon/extcon-class.c
index af03c9cc7818..dcbd857b8276 100644
--- a/drivers/extcon/extcon-class.c
+++ b/drivers/extcon/extcon-class.c
@@ -53,7 +53,10 @@ const char extcon_cable_name[][CABLE_NAME_MAX + 1] = {
 	[EXTCON_CDP]		= "CHARGER_USB_CDP",
 	[EXTCON_ACA]		= "CHARGER_USB_ACA",
 	[EXTCON_AC]		= "CHARGER_AC",
-	[EXTCON_TYPEC]		= "USB_TYPEC",
+	[EXTCON_TYPEC_DFP]	= "USB_TYPEC_DFP",
+	[EXTCON_TYPEC_UFP]	= "USB_TYPEC_UFP",
+	[EXTCON_TYPEC_DP_SOURCE]	= "USB_TYPEC_DP_SOURCE",
+	[EXTCON_TYPEC_DP_SINK]	= "USB_TYPEC_DP_SINK",
 	[EXTCON_HDMI]		= "HDMI",
 	[EXTCON_MHL]		= "MHL",
 	[EXTCON_DVI]		= "DVI",
diff --git a/include/linux/extcon.h b/include/linux/extcon.h
index 3e4d4f886580..2b3cf2a35b08 100644
--- a/include/linux/extcon.h
+++ b/include/linux/extcon.h
@@ -61,7 +61,10 @@ enum extcon_cable_name {
 	EXTCON_CDP,
 	EXTCON_ACA,
 	EXTCON_AC,
-	EXTCON_TYPEC,
+	EXTCON_TYPEC_DFP,
+	EXTCON_TYPEC_UFP,
+	EXTCON_TYPEC_DP_SOURCE,
+	EXTCON_TYPEC_DP_SINK,
 	EXTCON_HDMI,
 	EXTCON_MHL,
 	EXTCON_DVI,
-- 
1.9.1

